src/core/data/allocator.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "core/data/buffer.h"
+
+#include <unordered_map>
+
+namespace legate {
+
+class IntraTaskAllocator {
```

Comment:
I have second thoughts on the naming, and this really should be `ScopedAllocator`, as that's what this allocator does. Also, we should make a warning that allocations are valid only within the scope that this allocator is alive; i.e., as soon as this allocator goes out of scope and its destructor runs, memory space for the allocations can be recycled.

Reply:
Agreed. I did change the naming and added a parameter to control destruction behavior. What kind of warning are you thinking about? I suppose we don't want a runtime warning, right?

Reply:
By warning I meant adding a comment to this class.